### PR TITLE
fix(es/tsc-test): Handle tsx

### PR DESCRIPTION
**Pull request by magic-akari**
**Description:**

**BREAKING CHANGE:**

**Related issue (if exists):**

### CODE REVIEW

File: crates/swc/tests/tsc-references/checkJsxChildrenProperty12.1.normal.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 //// [file.tsx]
 //! 
-//!   x Expected '>', got '{'
-//!     ,----
-//!  14 | return <InnerButton {...this.props} />
-//!     :                     ^
-//!     `----
+//!   x Import assignment cannot be used when targeting ECMAScript modules. Consider using `import * as ns from "mod"`, `import {a} from "mod"`, `import d from "mod"`, or another module format instead.
+//!    ,----
+//!  2 | import React = require('react');
```
Comment (magic-akari) [inline crates/swc/tests/tsc-references/checkJsxChildrenProperty12.1.normal.js:5]: Maybe we should set `CommonJS` as default Module config.

Comment (magic-akari) [reply crates/swc/tests/tsc-references/checkJsxChildrenProperty12.1.normal.js:5]: Yes.

https://github.com/swc-project/swc/blob/390b1d12e30a30b4791f493e34bfc8b6a790c85d/crates/swc/tests/tsc.rs#L112-L114

and

https://github.com/swc-project/swc/blob/390b1d12e30a30b4791f493e34bfc8b6a790c85d/crates/swc/tests/tsc.rs#L175-L177

Comment (kdy1) [reply crates/swc/tests/tsc-references/checkJsxChildrenProperty12.1.normal.js:5]: I'm fine with changing the default

Comment (magic-akari) [reply crates/swc/tests/tsc-references/checkJsxChildrenProperty12.1.normal.js:5]: Hmmm. too many files changed. should not do it in this pull request.

File: crates/swc/tests/tsc-references/jsxParsingError2.1.normal.js [hunk 2]
```diff
@@ -2,37 +2,36 @@
 //// [file.tsx]
 //// [Error1.tsx]
 //! 
-//!   x Unexpected eof
+//!   x Unexpected token `div`. Expected jsx identifier
```
Comment (magic-akari) [inline crates/swc/tests/tsc-references/jsxParsingError2.1.normal.js:5]: confusing error message. We can improve it.

**Review by kdy1 — APPROVED** [review]
Thanks!

---

swc-bump:
 - swc

**Pull request merged**